### PR TITLE
Notifications and feedback on transactions

### LINKED ISSUE #106: Global transaction watcher (toast, notifications)
## Overview
There would be many opportunities to improve user experience if there were a global system for monitoring the status of a transaction triggered by the UI

## Acceptance criteria
- when the user stakes or claims unstaked s/tlos, a toast notification should be displayed saying what happened, eg. "Staked 1 TLOS for .3 sTLOS completed"
- functions invoked from the function interfaces in the contract tab, eg. [here](https://www.teloscan.io/address/0xB4B01216a5Bc8F1C8A33CD990A1239030E60C905#contract), should pop a toast when the transaction is completed

## Technical considerations
A global api which is given a transaction hash and other data, it then polls for the transaction to be executed and renders a toast to the user, and calls a callback from the component.

Possibly this could be integrated into a centralized "sendTransaction" function.

Data to pass into it:

Name: "sTLOS staking"
Completion message: "Staked 1 TLOS for .3 sTLOS completed"
Callback: function from sTLOS page that would reload balances

- when transactions initiated by the UI are completed, a toast should be displayed to the user
- there should also be the ability to trigger a callback, or possibly a singleton pub/sub pattern, to allow components to react to the completion of a transaction
- this solution should gracefully handle multiple transactions, showing toasts in succession when multiple complete in overlapping timeframes

### PR DESCRIPTION
**Pull request by Viterbo**
# Fixes #106

## Description
This PR includes a packed solution to simplify the task of notifying the user whenever a transaction is signed and sent, succeeded or not, with a simple call to a generic signCustomTransaction function.

## Test scenarios
- go to this link
- login
- go to Liquid Staking
  - stake, unstake and withdraw
  - in all cases, it should:
    - put the button in the loading state (and stop when done)
    - after confirming in the wallet, a neutral notification should say what is happening (auto dismissed)
    - if it succeeds, a green success notification is shown with a "see transaction" button redirecting to the transaction page
    - if any error occurs, a red error notification is shown with a "see details" button which opens a popup with the error JSON 
- go to a custom contract [like stlos contract](https://deploy-preview-466--teloscan-stage.netlify.app/address/0xa9991E4daA44922D00a78B6D986cDf628d46C4DD)
  - pick a function you know you can call (like approve)
  - fill in the form and press the write button
    - put the button in the loading state (and stop when done)
    - after confirming in the wallet, a neutral notification should say 'calling **foo** with **N** params'' (auto dismissed)
    - if it succeeds, a green success notification is shown with a "see transaction" button redirecting to the transaction page
    - if any error occurs, a red error notification is shown with a "see details" button which opens a popup with the error JSON 

## Screen captures

[Notifications-on-Teloscan.webm](https://github.com/telosnetwork/teloscan/assets/4420760/9ba2dfec-a07c-49c2-b8d3-bc4fcbb07cb2)

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages
-   [x] I have included all english text to the translation file and/or created a new issue with the required translations for the currently supported languages
-   [ ] I have tested for mobile functionality and responsiveness
-   [ ] I have added appropriate test coverage

### CODE REVIEW

**Review by ezra-sg — DISMISSED** [review]
awesome job, works great 💯

File: src/antelope/mocks/AntelopeConfig.ts [hunk 1]
```diff
@@ -16,12 +19,261 @@ export class AntelopeWallets {
...
 
-const config = {
-    notifyNeutralMessageHandler: (message: string) => void 0,
-    localizationHandler: (message: string, params?: Record<string, string>) => message,
-    notifyFailureWithActionHandler: (message: string, action: () => void) => void 0,
-    notifyFailureWithAction: (message: string, params?: { label: string; handler: () => void; }) => void 0,
-};
+export class AntelopeConfig {
+    // @TODO rename this method, it's used for token and NFT transfers as well
+    wrapError(description: string, error: unknown): AntelopeError {
```
Comment (donnyquixotic) [inline src/antelope/mocks/AntelopeConfig.ts:24]: rename to `transactionError` (I updated this in telos-wallet as part of the nft transfer PR)

File: src/antelope/mocks/AccountStore.ts [hunk 2]
```diff
@@ -19,34 +34,101 @@ interface LoginEVMActionData {
...
+                );
+            }).catch((err) => {
+                console.error(err);
+            }).finally(() => {
+                dismiss();
+            });
+
+            return tx;
+        } catch (error) {
+            const trxError = ant.config.wrapError(actionError, error);
```
Comment (donnyquixotic) [inline src/antelope/mocks/AccountStore.ts:122]: rename `wrapError` to `transactionError` see feedback in AntelopeConfig.ts

**Review by donnyquixotic — CHANGES_REQUESTED** [review]

**Comment by donnyquixotic** [comment]
@Viterbo is there a reason why all the new store files are added under the 'mocks' dir instead of 'store'?

**Comment by Viterbo** [comment]
> @Viterbo is there a reason why all the new store files are added under the 'mocks' dir instead of 'store'?

Yes. It's because they are mocks :laughing: 
No, out of joke, those clases are not the actual Stores. They are just simplifyed and adapted version of the same Store-API we have in the real Antelope code (back in telos-wallet repo).

Why did I do that? because otherwise I would need to migrate the whole library and that would be an enormous effort too early. The job needs to be done but not now, rather after the lib is finished

**Review by donnyquixotic — DISMISSED** [review]

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**